### ISSUE TITLE
Change recent-code to navigate on a per tab basis. Bonus: navigate in hovered tab.

### ISSUE THREAD

**Issue by CodingMeSwiftly** [seivan/swipe-to-navigate#12]
Love the extension! 
I use the recent-code setting and I have one single source of pain with it: I usually work with two tabs, side by side and the extension (or maybe VS code itself) does not keep track of navigation history on a per tab basis but rather globally. So often I want to navigate back or forward in the context of a given tab, but what I end up with is navigating back to the last location regardless of tab. For example, when I switch focus between tabs a couple times, swiping back will backtrack the focus switches and effectively only navigate to a recent code location until all the context switches are iterated through.
So I guess in summary my feature request is:
- Don't count tab focus switches as navigation events.
- Swipe to navigate to last/next location in the active tab, not globally.
- Bonus: Choose tab to trigger the navigation in by mouse location (hovered tab will navigate).

The recent-code setting says 'like Xcode' in its description, but the behavior described above is something that is missing to have trackpad navigation behavior 1:1 like Xcode.

Do you think this is something this extension could support?

**Comment by CodingMeSwiftly** [comment]
Is there any ETA on this? :) I looked into it myself today, but unfortunately, I got stuck trying to debug the patches/.js files.

From what I've gathered, we should be able to get the swipe location from the event data [here](https://github.com/seivan/swipe-to-navigate/blob/8835f751a7a6fe07314365e683d170ffc5e45c6b/patches/swipe-to-navigate.js#L38) and map that location to a respective tab pane. We can then activate that tab using [VSCode API](https://code.visualstudio.com/api/references/vscode-api#3414) before dispatching the navigation action.

That's just my limited view on things, I'm not very experienced with VSCode extension programming so I don't really know if the approach is feasible.

Additionally, the most recent [VSCode update](https://code.visualstudio.com/updates/v1_65) changes tab navigation and I don't know if that affects the proposed solution further.

What I can say so far, is that the VSCode navigation update doesn't break anything currently implemented in this extension. At least for me..
